src/createSchemaForm.tsx

```diff
@@ -443,14 +494,51 @@ export function createTsForm<
         form.reset(defaultValues);
       }
     }, []);
-    const { control, handleSubmit, setError, getValues } = _form;
+    const { handleSubmit, setError } = _form;
     const submitter = useSubmitter({
       resolver,
       onSubmit,
       setError,
     });
     const submitFn = handleSubmit(submitter.submit);
 
+    return (
+      <FormProvider {..._form}>
+        <SubmitterContextProvider {...submitter}>
+          <FormComponent {...formProps} onSubmit={submitFn}>
+            <FormFragment
+              {...({
+                schema,
+                props,
+                renderAfter:
+                  renderAfter && (() => renderAfter({ submit: submitFn })),
+                renderBefore:
+                  renderBefore && (() => renderBefore({ submit: submitFn })),
+                children: CustomChildrenComponent,
+              } as any)}
+            />
+          </FormComponent>
+        </SubmitterContextProvider>
+      </FormProvider>
+    );
+  }
+
+  function FormFragment<SchemaType extends RTFFormSchemaType>({
+    schema,
+    props,
+    renderAfter,
+    renderBefore,
+    children: CustomChildrenComponent,
+    name,
+  }: RTFSharedFormProps<Mapping, SchemaType, PropsMapType> & {
+    renderBefore?: (props: { submit?: () => void }) => ReactNode;
+    renderAfter?: (props: { submit?: () => void }) => ReactNode;
+    name?: string;
```

Comment:
@iway1 here's where i accept the name prop, it's basically an intermediate name between the current field name and the keys of the schema we're rendering here.

So if you are rendering an array property called `items` as a custom component using this, the `prefixKey` would be `items` the name you would pass for each item as `[1]` or whatever the index of the item is and `stringKey` will wind up being the names of each of the keys of whatver object `item` represents.